lib/generators.nix

```diff
@@ -525,6 +525,8 @@ ${expr "" v}
           "(${v.expr})"
         else if v == { } then
           "{}"
+        else if libAttr.isDerivation v then
+          ''"${toString v}"''
```

Comment:
(not very important and can be ignored)

We can delegate it to `builtins.toJSON v`  like `isInt v` and `isString v` to ensure proper quoting and escaping. Though, for store paths it shouldn't require escapes.

P.S. I would assume that `''"${v}"''` would work the same as `''"${toString v}"''` .

Reply:
thanks for the review. I will merge this now to unblock other stuff but I keep in mind your comment for my next PRs, I might add some tests too.
I would also like if we could avoid nix-specific names and use directly in buildLuarocksPackages the luarocks config names. @mrcjkb